code/modules/research/experimentor.dm

```diff
@@ -722,6 +728,165 @@
+	playsound(src, 'sound/magic/cosmic_energy.ogg', vol = 50, vary = TRUE)
+	for(var/mob/living/carbon/nearby in hearers(2, src))
+		nearby.bioscramble(name)
+		playsound(get_turf(nearby), SFX_SPARKS, rand(25,50), TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+		throw_smoke(get_turf(nearby))
+		to_chat(nearby, span_notice("You feel weird."))
+	if(prob(33))
+		qdel(src)
+
+// Charges an item or two in your inventory. Also yourself.
+/obj/item/relic/proc/charger(mob/living/user)
+	to_chat(user, span_danger("You're recharged!"))
+	var/stunner = 1.25 SECONDS
+	if(iscarbon(user))
+		var/mob/living/carbon/carboner = user
+		carboner.electrocute_act(15, src, flags = SHOCK_NOGLOVES, stun_duration = stunner)
+	else
+		user.electrocute_act(15, src, flags = SHOCK_NOGLOVES)
+	playsound(get_turf(user), SFX_SPARKS, rand(25,50), TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+
+	var/list/chargeable_batteries = list()
+	for(var/obj/item/stock_parts/power_store/C in user.get_all_contents())
+		if(C.charge < (C.maxcharge * 0.95)) // otherwise the PDA always gets recharged
+			chargeable_batteries |= C
+
+	lightning_fx(user, stunner)
+	var/recharges = rand(1, 2)
+	if(!length(chargeable_batteries))
+		to_chat(user, span_notice("You have a strange feeling for a moment, but then it passes."))
+		return
+	for(var/obj/item/stock_parts/power_store/to_charge as anything in 1 to recharges)
+		to_charge = pick(chargeable_batteries)
+		to_charge.charge = to_charge.maxcharge
+		// The device powered by the cell is assumed to be its location.
+		var/obj/device = to_charge.loc
+		// If it's not an object, or the loc's assigned power_store isn't the cell, undo.
+		if(!istype(device) || (device.get_cell() != to_charge))
+			device = to_charge
+		device.update_appearance(UPDATE_ICON|UPDATE_OVERLAYS)
+		to_chat(user, span_notice("[device] feels energized!"))
+		lightning_fx(device, 0.8 SECONDS)
+
+/obj/item/relic/proc/lightning_fx(atom/shocker, time)
+	var/lightning = mutable_appearance('icons/effects/effects.dmi', "electricity3", layer = ABOVE_MOB_LAYER)
+	shocker.add_overlay(lightning)
+	addtimer(CALLBACK(src, PROC_REF(cut_the_overlay), shocker, lightning), time)
+
+/obj/item/relic/proc/cut_the_overlay(atom/shocker, lightning)
+	shocker.cut_overlay(lightning)
+
+// Hugs/shakes everyone in range!
+/obj/item/relic/proc/hugger(mob/user)
+	var/list/mob/living/carbon/huggeds = oviewers(3, user)
+	for(var/mob/living/carbon/victim in huggeds)
+		victim.help_shake_act(user, force_friendly = TRUE)
+		new /obj/effect/temp_visual/heart(victim.loc)
+	if(length(huggeds))
+		to_chat(user, span_nicegreen("You feel friendly!"))
+	else
+		to_chat(user, pick(span_notice("You hug yourself, for some reason."), span_notice("You have a strange feeling for a moment, but then it passes.")))
+
+// Converts a 3x3 area into a random dimensional theme.
+/obj/item/relic/proc/dimensional_shift(mob/user)
+	var/datum/dimension_theme/shifter = pick(subtypesof(/datum/dimension_theme))
+	shifter = new shifter(src)
```

Comment:
Pretty sure themes are singletons, you don't make new ones

Reply:
using uninstanced themes made it always turn things into iron. probably due to incorrect code

Reply:
Idk I see
`terrain_theme = SSmaterials.dimensional_themes[pick(subtypesof(/datum/dimension_theme))]`
so I don't think you should be `new`ing here